app/models/temp_site.rb

```diff
@@ -18,6 +18,10 @@ def self.for_recruitment_cycle_years(recruitment_cycle_years)
     .distinct
   end
 
+  def self.uniq_by_location
+    all.uniq { |site| site.slice(:longitude, :latitude) }
```

Comment:
Is it a problem to load all records into memory?

Reply:
If I understood correctly the `format_average_distance` only calls `#longitude` and `#latitude` on the object passed in the view.

Based on that can we do something like this?

```
  all.select(:longitude, :latitude).uniq
```

But happy as it is. This is minor thing.

Reply:
Thanks @tomas-stefano. TIL Rails doesn't support `DISTINCT ON`, so have just used SQL. I think this addresses the performance issue.